app/assets/scripts/components/slate/plugins/equation/index.js

```diff
@@ -39,15 +43,23 @@ const deleteEquation = (editor) => {
  *
  * @param {Editor} editor Slate editor instance.
  */
-export const insertEquation = (editor) => {
+export const upsertEquation = (editor, equation, nodePath) => {
   const node = {
     type: EQUATION,
-    children: [{ text: 'latex~example: e=mc^2' }]
+    children: [{ text: equation }]
   };
 
-  const path = getPathForRootBlockInsert(editor);
-  Transforms.insertNodes(editor, node, { at: path });
-  Transforms.select(editor, path);
+  if (nodePath) {
+    Transforms.removeNodes(editor, { at: nodePath });
+    Transforms.insertNodes(editor, node, { at: nodePath });
```

Comment:
Is there a different way to update, other than removing the node and inserting a new one?